server/util/rexec/rexec.go

```diff
@@ -0,0 +1,282 @@
+// package rexec provides a collection of utility functions for remote execution
+// clients.
+package rexec
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"sort"
+	"strings"
+
+	"github.com/buildbuddy-io/buildbuddy/server/environment"
+	"github.com/buildbuddy-io/buildbuddy/server/interfaces"
+	"github.com/buildbuddy-io/buildbuddy/server/remote_cache/cachetools"
+	"github.com/buildbuddy-io/buildbuddy/server/remote_cache/digest"
+	"github.com/buildbuddy-io/buildbuddy/server/util/retry"
+	"github.com/buildbuddy-io/buildbuddy/server/util/status"
+	"golang.org/x/exp/maps"
+	"golang.org/x/sync/errgroup"
+	"google.golang.org/genproto/googleapis/longrunning"
+
+	repb "github.com/buildbuddy-io/buildbuddy/proto/remote_execution"
+	rspb "github.com/buildbuddy-io/buildbuddy/proto/resource"
+	gstatus "google.golang.org/grpc/status"
+)
+
+// AssembleEnv assembles a list of EnvironmentVariable protos from a list of
+// NAME=VALUE pairs. If the same name is specified more than once, the last one
+// wins. The entries are sorted by name, so that the environment variables are
+// cache-friendly.
+func AssembleEnv(pairs ...string) ([]*repb.Command_EnvironmentVariable, error) {
```

Comment:
probably would call this MakeEnv or MakeEnvVars (and MakePlatform) below b/c Assemble is a little wordy

Reply:
Done